sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+    dataschema?: string;
+    id: string;
+    source: string;
+    subject?: string;
+    time?: Date;
+    type: string;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventData {
+    action?: string;
+    id?: string;
+    target?: ContainerRegistryArtifactEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventTarget {
+    digest?: string;
+    mediaType?: string;
+    name?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    version?: string;
+}
+
+// @public
+export type ContainerRegistryChartDeletedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export type ContainerRegistryChartPushedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export interface ContainerRegistryEventActor {
+    name?: string;
+}
+
+// @public
+export interface ContainerRegistryEventData {
+    action?: string;
+    actor?: ContainerRegistryEventActor;
+    id?: string;
+    request?: ContainerRegistryEventRequest;
+    source?: ContainerRegistryEventSource;
+    target?: ContainerRegistryEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryEventRequest {
+    addr?: string;
+    host?: string;
+    id?: string;
+    method?: string;
+    useragent?: string;
+}
+
+// @public
+export interface ContainerRegistryEventSource {
+    addr?: string;
+    instanceID?: string;
+}
+
+// @public
+export interface ContainerRegistryEventTarget {
+    digest?: string;
+    length?: number;
+    mediaType?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    url?: string;
+}
+
+// @public
+export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData & {};
+
+// @public
+export type ContainerRegistryImagePushedEventData = ContainerRegistryEventData & {};
+
+// @public
+export class EventGridClient {
+    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridClientOptions);
+    readonly apiVersion: string;
+    readonly endpointUrl: string;
+    generateSharedAccessSignature(expiresOnUtc: Date): string;
+    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
+    sendCustomSchemaEvents(events: Record<string, any>[], options?: SendCustomSchemaEventsOptions): Promise<RestResponse>;
+    sendEvents(events: EventGridEvent<any>[], options?: SendEventsOptions): Promise<RestResponse>;
+}
+
+// @public
+export type EventGridClientOptions = PipelineOptions;
+
+// @public
+export class EventGridConsumer {
+    constructor(decoders?: Record<string, CustomEventDataDecoder>);
+    // Warning: (ae-forgotten-export) The symbol "CustomEventDataDecoder" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    readonly customDecoders: Record<string, CustomEventDataDecoder>;
+    decodeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
```

Comment:
I have a vague worry about `unknown`, which is that I am unsure if even hardcore TS users prefer its safety to `any`. Any thoughts @xirzec?

Reply:
hmm unknown at least tells you to do some kind of casting, which in a sense can be somewhat preferable to `any`. 

Is the idea here that someone would iterate over this array and do a kind of isFooEvent(event) that could tell if `event is a FooEvent`?

Reply:
> Is the idea here that someone would iterate over this array and do a kind of isFooEvent(event) that could tell if event is a FooEvent?

Yes.